app/scripts/lib/account-tracker.js

```diff
@@ -97,20 +98,24 @@ export default class AccountTracker {
       }, this.onboardingController.store.getState()),
     );
 
-    this.preferencesController.store.subscribe(
-      previousValueComparator(async (prevState, currState) => {
-        const { selectedAddress: prevSelectedAddress } = prevState;
-        const {
-          selectedAddress: currSelectedAddress,
-          useMultiAccountBalanceChecker,
-        } = currState;
+    this.selectedAccount = this.controllerMessenger.call(
+      'AccountsController:getSelectedAccount',
+    );
+
+    this.controllerMessenger.subscribe(
+      'AccountsController:selectedAccountChange',
+      (newAccount) => {
+        const { useMultiAccountBalanceChecker } =
```

Comment:
What if the `useMultiAccountBalanceChecker` preference gets changed? It looks like previously accounts would get updated and now that would no longer happen.

Reply:
Maybe I'm missing something here too, but I see the same logic as before

Reply:
After taking a second look at this one I think the logic is correct because despite `useMultiAccountBalanceChecker` the logic inside the if only is executed if the selected address also changes which is not the case if only the `useMultiAccountBalanceChecker` value gets updated.

Here's the previous logic and let me know if it makes sense to you too,

```ts
this.preferencesController.store.subscribe(
  previousValueComparator(async (prevState, currState) => {
    const { selectedAddress: prevSelectedAddress } = prevState;
    const {
      selectedAddress: currSelectedAddress,
      useMultiAccountBalanceChecker,
    } = currState;
    if (
      prevSelectedAddress !== currSelectedAddress &&
      !useMultiAccountBalanceChecker
    ) {
      this._updateAccounts();
    }
  }, this.onboardingController.store.getState()),
);
```

Reply:
Oh gotcha. Yeah you're right, we didn't handle that before. Okay, this looks good.